spec/src/main/asciidoc/property-expression.adoc

```diff
@@ -0,0 +1,76 @@
+= Property Expression
+
+The property expression is in the format of ${a.property}. When resoving the property expression, the value of `a.property` will be used.
+With the property expression, it enables a property refer to another property.
```

Comment:
How does this relate syntactically to configuration sources? Are the raw values subject to multiple stages of parsing, such that property expression resolution is a layer above configuration sources? Or does each configuration source have a responsibility to independently specify a syntactic format and parsing strategy for a property value?

Reply:
It is not the config source that resolves the ultimate value, but it is the responsibility of the config that resolves the config value.

Reply:
(Of course, this specification has not defined "config" or "config source" or what it means to "resolve".  [This comment](https://github.com/eclipse-ee4j/config/pull/78#discussion_r865994897) supposes that there is an aggregate object ("config") that mediates among sub-objects ("config sources").  There may end up being no such object, or there may.)

David's question still stands: there is a proposed syntax in this PR that overlays a property expression mechanism on top of something.  Or maybe it does not.  It is unclear what this PR actually does as written IMHO.

I think [the comment above](https://github.com/eclipse-ee4j/config/pull/78#discussion_r865994897) is trying to say something like: loosely, property value substitution occurs "on top of" any sub-objects that this specification may define, i.e. the sub-objects don't do it, something else "above" them does (if we end up defining such an architecture).

If that ends up being true, then it is critical to rigorously define its syntax, which this PR does not yet do, so that we can convey to anyone who wants to write a sub-object implementation how to work with this mechanism.  Consider a configuration property whose value is a regex, for example, featuring lots of `$`, `\`, `{`, `[` and other awful characters.  What is escaped and how in such a case?  This has to be spelled out unambiguously so that people can get all the backslashes (or whatever) right.

Reply:
My question isn't about how the config sources relate to one another on a single config object, but rather how property expansion relates to the config object as a whole. If a config object comprises two config sources, and it is a child of an enclosing config object, it would be my expectation that property expansion would occur relative to the config object that the config source is registered to, not its parent (or child(ren)).